### DISCUSSION TITLE
Could you add a few more examples?

### DISCUSSION THREAD

**Discussion by sproctor** [evant/kotlin-inject#207] [General]
I don't see an obvious way to extract a provided value from a component. I have something like:

```kotlin
@Component
abstract class DataComponent {
    @Provides
    fun provideRepo(impl: RepoImpl): Repo = impl
}

@Inject
class RepoImpl(...) { ... }
```

I would like to get a Repo from the component, but I don't see how. I guess I could do something like: `abstract val repo: Repo`?

Is there a way to inject part of a constructor? In Dagger: https://dagger.dev/dev-guide/assisted-injection.html

**Comment by evant** [comment]
> I would like to get a Repo from the component, but I don't see how. I guess I could do something like: abstract val repo: Repo?

Yes

> Is there a way to inject part of a constructor?

https://github.com/evant/kotlin-inject#function-support describes how you do assisted injection

**Reply by sproctor** [reply]
Ah, excellent, thank you. I had just read "function support" and skipped reading that section to my own detriment.

Thanks so much for the library. My path from dagger/hilt -> manual -> this is complete. I'm ready to port to other platforms now.

**Comment by evant** [comment]
Updated the section title to make it more clear hopefully
